homeassistant/components/alarm_control_panel/group.py

```diff
@@ -0,0 +1,241 @@
+"""
+Group platform for alarm control panel component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/alarm_control_panel.group/
+"""
+import asyncio
+from collections import Counter
+from copy import deepcopy
+import logging
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.components.alarm_control_panel import (
+    DOMAIN, SERVICE_ALARM_DISARM, SERVICE_ALARM_ARM_HOME,
+    SERVICE_ALARM_ARM_AWAY, SERVICE_ALARM_ARM_NIGHT, SERVICE_ALARM_TRIGGER,
+    SERVICE_ALARM_ARM_CUSTOM_BYPASS)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_PRE_PENDING_STATE, ATTR_POST_PENDING_STATE,
+    STATE_ALARM_DISARMED, STATE_ALARM_ARMED_CUSTOM_BYPASS, STATE_ALARM_PENDING,
+    STATE_ALARM_TRIGGERED, CONF_PLATFORM, CONF_NAME, EVENT_HOMEASSISTANT_START)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_state_change
+
+ATTR_TRIGGERED_PANELS = 'triggered_panels'
+
+CONF_CODE_FORMAT = 'code_format'
+CONF_PANELS = 'panels'
+CONF_PANEL = 'panel'
+
+DEFAULT_ALARM_NAME = 'Group Alarm'
+
+PLATFORM_SCHEMA = vol.Schema(vol.All({
+    vol.Required(CONF_PLATFORM): 'group',
+    vol.Optional(CONF_NAME, default=DEFAULT_ALARM_NAME): cv.string,
+    vol.Optional(CONF_CODE_FORMAT, default=''): cv.string,
+    vol.Required(CONF_PANELS): vol.All(cv.ensure_list, [{
+        vol.Required(CONF_PANEL): cv.entity_id,
```

Comment:
The `CONF_PANEL` key is mostly for future extensibility, in case we want to add more panel-specific configuration.

Having the same entity_id multiple times would probably be a bug in the configuration. However, I haven't seen uses of `vol.Unique` in hass, which is why I left it out.

Reply:
I think something like the scene config makes more sense. But right now there are no values for the entity keys:
https://home-assistant.io/components/scene/.